pkg/generate/crds/gen.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crds
```

Comment:
We could structure the pkgs as follows:

`pkg/internal/parse` --> parse related stuff but for internal consumption only
`pkg/crd/` --> everything related to crd
`pkg/crd/generator` --> CRD generation related code goes here ? 

I found this guide useful for best practices for pkgs --> https://rakyll.org/style-packages/